server/broadcast_test.go

```diff
@@ -451,6 +460,115 @@ func TestCleanupSessions(t *testing.T) {
 //     assert an error from transcoder removes sess from BroadcastSessionManager
 //     assert a success re-adds sess to BroadcastSessionManager
 
+func TestTranscodeSegment_ExpiredParams_GetOrchestratorInfoAndRetry(t *testing.T) {
+	require := require.New(t)
+	assert := assert.New(t)
+
+	// Create stub server
+	ts, mux := stubTLSServer()
+	defer ts.Close()
+
+	tr := &net.TranscodeResult{
+		Info: &net.OrchestratorInfo{
+			Transcoder:   ts.URL,
+			PriceInfo:    &net.PriceInfo{PricePerUnit: 7, PixelsPerUnit: 7},
+			TicketParams: &net.TicketParams{ExpirationBlock: big.NewInt(100).Bytes()},
+		},
+		Result: &net.TranscodeResult_Data{
+			Data: &net.TranscodeData{
+				Segments: []*net.TranscodedSegmentData{&net.TranscodedSegmentData{Url: "test.flv"}},
+				Sig:      []byte("bar"),
+			},
+		},
+	}
+	buf, err := proto.Marshal(tr)
+	require.Nil(err)
+
+	mux.HandleFunc("/segment", func(w http.ResponseWriter, r *http.Request) {
+		w.WriteHeader(http.StatusOK)
+		w.Write(buf)
+	})
+
+	sess := StubBroadcastSession(ts.URL)
+	sender := &pm.MockSender{}
+	sess.Sender = sender
+	balance := &mockBalance{}
+	sess.Balance = balance
+	sess.Profiles = []ffmpeg.VideoProfile{ffmpeg.P144p30fps16x9}
+	sess.OrchestratorInfo = &net.OrchestratorInfo{
+		Transcoder: ts.URL,
+	}
+
+	cxn := &rtmpConnection{
+		mid:         core.ManifestID("foo"),
+		nonce:       7,
+		pl:          &stubPlaylistManager{manifestID: core.ManifestID("foo")},
+		profile:     &ffmpeg.P144p30fps16x9,
+		sessManager: bsmWithSessList([]*BroadcastSession{sess}),
+	}
+
+	// Expired Orchestrator Info -> GetOrchestratorInfo error -> Error
+	sender.On("ValidateTicketParams", mock.Anything).Return(pm.ErrTicketParamsExpired)
+	_, err = transcodeSegment(cxn, &stream.HLSSegment{Data: []byte("dummy"), Duration: 2.0}, "dummy", nil)
+	assert.True(strings.Contains(err.Error(), "unable to refresh ticketparams"))
+
+	// Expired Orchestrator Info -> GetOrchestratorInfo -> Still Expired -> Error
+	cxn = &rtmpConnection{
+		mid:         core.ManifestID("bar"),
+		nonce:       8,
+		pl:          &stubPlaylistManager{manifestID: core.ManifestID("bar")},
+		profile:     &ffmpeg.P144p30fps16x9,
+		sessManager: bsmWithSessList([]*BroadcastSession{sess}),
+	}
+
+	successOrchInfoUpdate := &net.OrchestratorInfo{
+		Transcoder: ts.URL,
+		PriceInfo: &net.PriceInfo{
+			PricePerUnit:  1,
+			PixelsPerUnit: 1,
+		},
+		TicketParams: &net.TicketParams{},
+	}
+
+	getOrchestratorInfoRPC = func(ctx context.Context, bcast common.Broadcaster, orchestratorServer *url.URL) (*net.OrchestratorInfo, error) {
```

Comment:
Let's make sure to reset `getOrchestratorInfoRPC` after the test function returns so that other tests that depend on the variable start with a clean slate. Ex.

```golang
oldGetOrchestratorInfoRPC = getOrchestratorInfoRPC
defer func () { getOrchestratorInfoRPC = oldGetOrchestorInfoRPC }()
...
```

Reply:
fixed in 20340ceabc68452fb4c8914a568e965e1c5f5d30